### ISSUE TITLE
Replace Reflectable transformer with Dart's new build system

### ISSUE THREAD

**Issue by jbaxe2** [PointyCastle/pointycastle#131]
Somewhat related to issue #119.  Version 1.0.0-rc2 supports Dart 2 semantics, yet still relies on using the transformer for the Reflectable package.  This should be updated to Dart's new build system.

Doing so would cause a need for the Reflectable package to be updated to at least 2.0.0-dev.1.0 (though 2.0.1 is current version), which may be more of a pain than at first glance.

**Comment by mathieujobin** [comment]
I hit an error today about Reflectable `Bad state: Reflectable has not been initialized` I wonder if its related?

**Comment by stevenroose** [comment]
Yeah lately I'm thinking about removing the whole reflection part (Dart doesn't seem to really like reflection) and revert to the good old registry with hard-coded values. If only I had the time..

**Comment by mathieujobin** [comment]
I'm getting the error using `Digest("MD5").process(uint8listvar);` any workaround ?

**Comment by mathieujobin** [comment]
I see, using `MD5Digest()` I don't have the same problem...

**Comment by mathieujobin** [comment]
if I am understanding this Reflectable transformer thing properly.. 
is this how `CBCBlockCipher(AESFastEngine())` is equivalent to `BlockCipher("AES/CBC")` ?

**Comment by mathieujobin** [comment]
Woohooo, I know this is not the place, but I got everything working. Thank you PointyCastle && Steven

**Comment by yshrsmz** [comment]
@stevenroose Hi, what's your plan for the hard-coded registry?
I've brought the code from https://github.com/PointyCastle/pointycastle/commit/2da75e5a8d7bdbf95d08329add9f13b9070b75d4 to my fork, and it looks like working.
If that's the plan, I'm going to send a PR with the change.

**Comment by stevenroose** [comment]
@yshrsmz I would remove reflactable (or leave it somewhere, but might cause problems having it in the pubspec even if most users don't use it) and create a new registry that just has a method that registers all implementations from within the codebase. This method would be called the first time the registry is used.

That would require users that want to use their own implementations to call the registry manually as well, which is a fair tradeoff for not having the reflection mess. (Right now, you can just define an implementation, like `class CrazyDigest implements Digest {` and import it and then you could use `new Digest("CRAZY")`. Without reflection, you'd have to do something like `AlgoRegistry<Digest>.register(CrazyDigest.FACTORY)`.

**Comment by stevenroose** [comment]
But yes, @yshrsmz , please create PRs for all the changes you made in your fork. I don't really have time to devote to this project, but I do want to keep it usable and I'm open to collaboration for maintaining it!
